torchdp/compute_dp-sdg_privacy.py

```diff
@@ -0,0 +1,139 @@
+    eps, opt_order = get_privacy_spent(orders, rdp, delta=delta)
+
+    print(
+        "DP-SGD with\n\tsampling rate = {:.3g}% and\n\tnoise_multiplier = {}"
+        "\n\titerated over {} steps\n satisfies".format(100 * q, sigma, steps),
+        end=" ",
+    )
+    print(
+        "differential privacy with\n\teps = {:.3g} and\n\tdelta = {}.".format(
+            eps, delta
+        )
+    )
+    print("The optimal RDP order is {}.".format(opt_order))
+
+    if opt_order == max(orders) or opt_order == min(orders):
+        print(
+            "The privacy estimate is likely to be improved by expanding "
+            "the set of orders."
+        )
+
+    return eps, opt_order
+
+
+def compute_dp_sgd_privacy(n, batch_size, noise_multiplier, epochs, delta):
+    """Compute epsilon based on the given hyperparameters."""
+    q = batch_size / n  # q - the sampling ratio.
+    if q > 1:
+        raise ValueError("n must be larger than the batch size.")
+    orders = (
+        [1.25, 1.5, 1.75, 2.0, 2.25, 2.5, 3.0, 3.5, 4.0, 4.5]
+        + list(range(5, 64))
+        + [128, 256, 512]
+    )
+    steps = int(math.ceil(epochs * n / batch_size))
+
+    return apply_dp_sgd_analysis(q, noise_multiplier, steps, orders, delta)
+
+
+def main():
+    # Settings w.r.t. parameters on command line or default values if missing
+    parser = argparse.ArgumentParser(description="RDP computation")
+    parser.add_argument(
+        "-n",
+        "--dataset-size",
+        type=int,
+        default=60_000,
+        help="Training dataset size (default: 60_000)",
+    )
+    parser.add_argument(
+        "-b",
+        "--batch-size",
+        type=int,
+        default=256,
+        help="Input batch size (default: 256)",
+    )
+    parser.add_argument(
+        "-s",
+        "--sigma",
+        type=float,
+        default=1.12,
+        help="Noise multiplier (default: 1.12)",
+    )
+    parser.add_argument(
+        "-e",
+        "--epochs",
+        type=int,
+        default=60,
+        help="Number of epochs to train (default: 60)",
+    )
+    parser.add_argument(
+        "-d",
+        "--delta",
+        type=float,
+        default=1e-5,
+        help="Targeted delta (default: 1e-5)",
+    )
+    args = parser.parse_args()
+
+    print("==================================================================")
+    print(
+        f"* Script called with those arguments:\n\t"
+        f"-n = --dataset-size = {args.training_dataset_size}\n\t"
+        f"-b = --batch-size = {args.batch_size}\n\t"
+        f"-s = --sigma = {args.sigma}\n\t"
+        f"-e = --epochs = {args.epochs}\n\t"
+        f"-d = --delta = {args.delta}\n"
+    )
+
+    print("* Result is", end=" ")
+    compute_dp_sgd_privacy(
+        args.training_dataset_size,
+        args.batch_size,
+        args.sigma,
+        args.epochs,
+        args.delta,
+    )
+    print("==================================================================")
```

Comment:
I don't know how it is possible, but I let a bug (I was sure to have checked carefully, I am absolutely sorry, please be indulgent, I'm still newbie wit git versioning...)
[`training_dataset_size` has to be replace by `dataset_size`]
```suggestion
   print("=================================================================="
          "\n* Script called with those arguments:\n\t"
          f"-n = --dataset-size = {args.dataset_size}\n\t"
          f"-b = --batch-size = {args.batch_size}\n\t"
          f"-s = --sigma = {args.sigma}\n\t"
          f"-e = --epochs = {args.epochs}\n\t"
          f"-d = --delta = {args.delta}\n"
          "* Result is:",
          end="\n  "
    )
    compute_dp_sgd_privacy(
        args.dataset_size,
        args.batch_size,
        args.sigma,
        args.epochs,
        args.delta,
    )
    print("==================================================================")
```

Reply:
Shame on me. But I agree with the solution I proposed...

Reply:
So I still have to modify wrong argument as I notified. But it's impossible for now (maybe because I modified the file name)